test/unit/auth-handlers.test.js

```diff
@@ -418,15 +418,98 @@ describe('Auth Handler:', function () {
     });
 
     describe('oauth2', function () {
-        it('should work correctly', function () {
-            var request = new Request(rawRequests.oauth2),
-                auth = request.auth,
-                authInterface = createAuthInterface(auth),
-                handler = AuthLoader.getHandler(auth.type);
+        var requestObj = {
+            auth: {
+                type: 'oauth2',
+                oauth2: {
+                    accessToken: '123456789abcdefghi',
+                    addTokenTo: 'header',
+                    tokenType: 'bearer'
+                }
+            },
+            url: 'https://api.github.com/user/orgs',
+            method: 'GET'
+        };
+
+        it('should sign the request by adding the token to the header', function () {
```

Comment:
Also, add the corresponding logic and unit test for adding params to the request body.

Reply:
params: see [line 448](https://github.com/postmanlabs/postman-runtime/pull/384/files#diff-ae634fae9ced1b620809db816fdab229R448)
body: not supported :(

Reply:
https://github.com/postmanlabs/postman-app-support/issues/3239

Reply:
The said issue is about sending the `client_id` and `client_secret` in the header/body as part of 2nd request in Oauth2 to get the `access_token`
This PR is about using `access_token` for a protected resource

The rules for the token itself are stated here https://tools.ietf.org/html/rfc6750#section-2.1
Although bearer token can also be sent in the body, the RFC says that supporting it in the header is a MUST for server and have many conditions for a client to be able to add in the body.